### ISSUE TITLE
[CT-2323] [Bug] Transferring ownership via meta in project.yml does not work

### ISSUE THREAD

**Issue by aBBDnGus** [dbt-labs/dbt-core#7218]
### Is this a new bug in dbt-core?

- [X] I believe this is a new bug in dbt-core
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

### Current Behavior

I want to transfer the ownership of my views to "my_role" after creating and I'm defining this in `project.yml` as follows:
```
models:
    +meta:
      owner: "my_role"
```
This does not change the ownership of the views. The view owner is still the user with which I am logged in.
I can see in the logs, that there is no command for changing ownership. Also with defining `{{ config(meta = {'owner': 'my_role'}) }}`, the ownership does not cange.

### Expected Behavior

With using this code snippet, the owner of the view should be changed in "my_role". A query like `alter view {{ this }} owner to my_role` should be executed.

### Steps To Reproduce

1. Add ```
models:
    +meta:
      owner: "my_role"
``` to your `project.yml`
2. Execute `dbt run`
3. Check table owner. This will be the owner you are logged in.

### Environment

```markdown
- OS: Windows 10
- Python: 3.9.12
- dbt-core: 1.4.5
- dbt-postgres: 1.4.5
```

### Which database adapter are you using with dbt?

postgres

**Comment by dbeatty10** [comment]
Thanks for reaching out @aBBDnGus !

### TLDR

If you want to change the `owner` of an object within the database, you will need to do that explicitly. One way to do so would be using a [`post_hook`](https://docs.getdbt.com/docs/build/hooks-operations#examples-using-hooks). This would be applicable if each object is intended to have a different owner.

Another way to do it would be to run dbt as that owner. This would be applicable if you want all objects to have the same owner.

Overall, I'd recommend taking a look at the [grants](https://docs.getdbt.com/reference/resource-configs/grants) configuration options and considering if your requirements can be met by just assigning grants to the relevant users or roles instead of needing to change the owner.

### More detail

The `meta` field doesn't affect any built-in behavior of dbt-core (such as assigning the `owner` of an object within a database). Rather, it's [purpose](https://docs.getdbt.com/reference/resource-configs/meta) is to set metadata, and it is viewable in two places:
- compiled into the `manifest.json` file generated by dbt
- viewable in the auto-generated documentation (`dbt docs generate` and `dbt docs serve`)

The spirit of the [`owner` in meta](https://docs.getdbt.com/reference/resource-configs/meta#designate-a-model-owner) example is the same as the [`owner` in exposures](https://docs.getdbt.com/docs/build/exposures) (which is to say just metadata not used by dbt-core itself).

I'm going to close this as `wontfix` since we don't intend for `meta` to be used to assign owners of database objects within a database.

**Comment by aBBDnGus** [comment]
Thanks for your extensive and helpful answer, @dbeatty10

Nevertheless, do you plan to tackle this general issue in an improvement? We are a team of multiple developers, where each has its own database user (shared accounts are not recommended for security reasons). If then one developer creates a view and doesn't change the owner to the common role, the other developers can't change or delete this anymore. Therefore it isn't sufficient to just use grants.
As this is a quite common case, can you think of an easier solution for this instead of explicitly execute a post-hook?

**Comment by dbeatty10** [comment]
@aBBDnGus to handle this issue in general we recommend the following:
- using [separate schemas per developer](https://docs.getdbt.com/docs/collaborate/environments#how-do-i-maintain-different-environments-with-dbt)

If one developer creates a view with this setup, it will only be in their own development schema, and it would not affect other developers.

### More detail

Many dbt set ups look like the following:

- separate [targets](https://docs.getdbt.com/docs/core/connection-profiles#understanding-targets-in-profiles) per environment (dev, prod, ci, etc) in your [`profiles.yml`](https://docs.getdbt.com/docs/core/connection-profiles)
- one shared database with one schema per developer
- within the database itself:
    - separate database users per developer
    - a separate database user for your production deployment

### Environment variables, FTW

Making use of [environment variables](https://docs.getdbt.com/reference/dbt-jinja-functions/env_var) within `profiles.yml` aligns with the [twelve-factor app methodology](https://12factor.net/).

`profiles.yml`
```yaml
your_project_name:
  target: dev
  outputs:
    dev:
      type: postgres
      user: "{{ env_var('DBT_USER') }}"
      schema: "{{ env_var('DBT_SCHEMA') }}"
      ...
    prod:
      type: postgres
      user: "{{ env_var('DBT_USER') }}"
      schema: "{{ env_var('DBT_SCHEMA') }}"
      ...
```

Two developers using their laptops would each have a different value for the `DBT_USER` environment variable. And the production environment would also have a different value. Likewise for `DBT_SCHEMA`.

### Environment variable examples

You:
```shell
export DBT_USER= abbdngus
export DBT_SCHEMA=dbt_abbdngus
```

Me:
```shell
export DBT_USER=dbeatty10
export DBT_SCHEMA=dbt_dbeatty10
```

Production environment:
```shell
export DBT_USER=prod_username
export DBT_SCHEMA=analytics
```

To build in the production environment would be:
```shell
dbt build --target prod
```

(Of course choosing the prod target should only be done by the `prod_dbt_user` user in the prod environment.)

Since `--target dev` is the default in the profiles.yml above, to build in a dev environment for either you or me would just be:
```shell
dbt build
```

### Advanced topic: custom schema configuration

- [How does dbt generate a model's schema name?](https://docs.getdbt.com/docs/build/custom-schemas#how-does-dbt-generate-a-models-schema-name)
- [Changing the way dbt generates a schema name](https://docs.getdbt.com/docs/build/custom-schemas#changing-the-way-dbt-generates-a-schema-name)
- [An alternative pattern for generating schema names](https://docs.getdbt.com/docs/build/custom-schemas#an-alternative-pattern-for-generating-schema-names)
